checksum.go

```diff
@@ -285,6 +286,58 @@ func (c *Client) ChecksumFromFile(ctx context.Context, checksumURL, checksummedP
 	return nil, fmt.Errorf("no checksum found in: %s", checksumURL)
 }
 
+// GetChecksum extracts the checksum from the `checksum` parameter
+// of the src of the Request
+// ex:
+//  http://hashicorp.com/terraform?checksum=<checksumValue>
+//  http://hashicorp.com/terraform?checksum=<checksumType>:<checksumValue>
+//  http://hashicorp.com/terraform?checksum=file:<checksum_url>
+// when the checksum is in a file, GetChecksum will first client.Get it
+// in a temporary directory, parse the content of the file and finally delete it.
+// The content of a checksum file is expected to be BSD style or GNU style.
+// For security reasons GetChecksum does not try to get the current working directory
+// and as a result, relative files will only be found when Request.Pwd is set.
+//
+// BSD-style checksum:
+//  MD5 (file1) = <checksum>
+//  MD5 (file2) = <checksum>
+//
+// GNU-style:
+//  <checksum>  file1
+//  <checksum> *file2
+func (c *Client) GetChecksum(ctx context.Context, req *Request) (*FileChecksum, error) {
```

Comment:
Nice !!!
I am just passing by - not sure if this is done and I'm sorry to bother if this was already planned - but I wanted to say make sure to put that code to replace the `extractChecksum` func so we don't keep it as dead code 🙂.